client/layout/masterbar/item.tsx

```diff
@@ -191,12 +190,8 @@ class MasterbarItem extends Component< MasterbarItemProps > {
 		}
 
 		// Check refs to see if the touch event started inside our component, if it didn't, close the menu.
-		const isInComponentButtonRef = this.componentButtonRef.current?.contains(
-			event.target as Node
-		);
-		const isInComponentDivRef = this.componentDivRef.current?.contains( event.target as Node );
-
-		if ( ! isInComponentButtonRef && ! isInComponentDivRef ) {
```

Comment:
Related to the comment above, since the wrapper `button` was removed, the logic can be simplified to only check whether we're in the wrapper div (which is now always rendered)